loom/script/native/core/system/lmDate.cpp

```diff
@@ -18,33 +18,102 @@
 * ===========================================================================
 */
 
+#include "loom/script/loomscript.h"
 #include "loom/script/native/lsLuaBridge.h"
 #include <ctime>
 
 using namespace LS;
 
+// Define a macro for getters and setters. There are a bunch of them!
+#define DATE_GETTER_SETTER(var, tm_var) \
+int get ## var() { return timeInfo.tm_ ## tm_var; } \
+void set ## var(int value) { timeInfo.tm_ ## tm_var = value; } \
+int get ## var ## UTC() { return timeInfoUTC.tm_ ## tm_var; } \
+void set ## var ## UTC(int value) { timeInfoUTC.tm_ ## tm_var = value; }
+
 class Date {
```

Comment:
The class should have docblock; probably it is sufficient to indicate its broad purpose then tell users to check the LoomScript side for fuller docs.